initial commit

samuel-oci opened the pull request:
### Description
Initial commit for the OCI object storage repository plugin.
 
### Issues Resolved
Initial implementation for OCI object storage repository plugin.
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

dblock commented on README.md:
````diff
@@ -1,11 +1,158 @@
+```
+Install the plugin on your Elasticsearch cluster
+```bash
+OS_HOME=<YOUR OPENSEARCH INSTALLATION PATH HERE> # (e.g. /Users/saherman/opensearch)
````
Recommend expanding `OS` into `OPENSEARCH` because OS is too operating system-y.

dblock commented on oci-repository-plugin/README.md:
```diff
@@ -0,0 +1,151 @@
+# Build and install the plugin
```
This duplicates parent README?

samuel-oci replied:
Looks like it slipped through the cracks, removed the duplicated README with the old references to ES.

dblock reviewed (commented):
Glad to see this coming in! I did a cursory review. 

1. Looks like you have some duplication and leftover Elasticsearch in READMEs. 
2. I recommend adding CI in this first PR and making sure the source code builds and runs tests from the start.
3. Consider explaining what this project is in README, and moving some of the existing README content into DEVELOPER_GUIDE.

samuel-oci commented:
Thank you for the review @dblock !

> Glad to see this coming in! I did a cursory review.
> 
>     1. Looks like you have some duplication and leftover Elasticsearch in READMEs.
removed
> 
>     2. I recommend adding CI in this first PR and making sure the source code builds and runs tests from the start.
That would be a great idea! I would have to wait until I get the admin permissions on the repo. Looks like github doesn't let me setup any actions at the moment.
> 
>     3. Consider explaining what this project is in README, and moving some of the existing README content into DEVELOPER_GUIDE.
Added explanation about the plugin in README, I also like the idea of separating it to DEVELOPER_GUIDE.

dblock commented:
If you PR a new CI GHA I think I can click a button to kick it off for you while you wait for access.

samuel-oci commented:
> If you PR a new CI GHA I think I can click a button to kick it off for you while you wait for access.

@dblock added a CI looks like it's all passing.

samuel-oci commented:
> > If you PR a new CI GHA I think I can click a button to kick it off for you while you wait for access.
> 
> @dblock added a CI looks like it's all passing.

Hi @dblock let me know if you feel it's good to merge at this point. The CI check also runs all the tests which seems to be passing.
